src/common/src/array/stream_chunk_iter.rs

```diff
@@ -12,17 +12,14 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::iter::once;
-
-use auto_enums::auto_enum;
-
 use super::data_chunk_iter::DataChunkRefIter;
+use super::stream_record::Record;
 use super::RowRef;
 use crate::array::{Op, StreamChunk};
 
 impl StreamChunk {
     /// Return an iterator on stream records of this stream chunk.
-    pub fn records(&self) -> StreamChunkRefIter<'_> {
+    pub fn records(&self) -> impl Iterator<Item = Record<RowRef<'_>>> {
```

Comment:
By convention, we should use concrete type here. 🤔 May take `itertools` and `futures` for examples.